iot/google/cloud/iot_v1/gapic/device_manager_client.py

```diff
@@ -769,11 +783,13 @@ def update_device(self,
             metadata = []
         metadata = list(metadata)
         try:
-            routing_header = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
-                [('device.name', device.name)], )
-            metadata.append(routing_header)
+            routing_header = [('device.name', device.name)]
```

Comment:
The same pattern as for `device_registry` above should apply here:  if `device` is a `dict`, then we should fetch its `name` key.